runtime/src/accounts_db.rs

```diff
@@ -4953,10 +4953,8 @@ impl AccountsDb {
     }
 
     pub fn generate_index(&self) {
-        // BTreeMap because we want in-order traversal of oldest write_version to newest.
-        // Thus, all instances of an account in a store are added to the index in oldest to newest
-        // order and we update refcounts and track reclaims correctly.
-        type AccountsMap<'a> = HashMap<Pubkey, BTreeMap<u64, (AppendVecId, StoredAccountMeta<'a>)>>;
+        type AccountsMap<'a> =
+            HashMap<Pubkey, (StoredMetaWriteVersion, AppendVecId, StoredAccountMeta<'a>)>;
```

Comment:
I think the old versioned accounts still need to be inserted into the index to properly update the refcount, https://github.com/solana-labs/solana/blob/master/runtime/src/accounts_index.rs#L204. This is necessary as some of the older entries in old storage entries may be preventing zero lamport keys from being cleaned.

Reply:
Discussed with @jeffwashington offline, this should be ok as the refcount is only incremented once per pubkey per slot anyways